include/zephyr/drivers/led_strip.h

```diff
@@ -110,6 +111,16 @@ static inline int led_strip_update_rgb(const struct device *dev,
 	const struct led_strip_driver_api *api =
 		(const struct led_strip_driver_api *)dev->api;
 
+	/* Allow for out-of-tree drivers that do not have this function for 2 Zephyr releases
+	 * until making it mandatory, function added after Zephyr 3.6
+	 */
+	if (api->length) {
+		/* Ensure supplied pixel size is valid for this device */
+		if (api->length(dev) < num_pixels) {
+			return -ERANGE;
+		}
+	}
+
```

Comment:
You can now remove the redundant check in `tlc5971_update_rgb()`.

And maybe the `num_pixels_ok()` check can be removed in `ws2812_strip_update_rgb()` as well.

Reply:
Removed the tlc5971 check, kept the ws2812 check as it's checking the buffer in regards to number of channels also

Reply:
The size of the internal buffer size used by the `ws2812_spi` is set to `cfg->num_colors * pixel-chain-length * 8`. Since `->update_rgb()` only copies `cfg->num_colors` for each passed pixel, then verifying the chain length is enough to ensure that the buffer is big enough. So `num_pixels_ok()` can be removed.

Reply:
Removed